ui/packages/app/web/src/pages/layouts/ThemeProvider.tsx

```diff
@@ -13,13 +13,50 @@
 
 import React, {useEffect} from 'react';
 
-import {selectDarkMode, useAppSelector} from '@parca/store';
+import {selectDarkMode, setDarkMode, useAppDispatch, useAppSelector} from '@parca/store';
 
 const ThemeProvider = ({children}: {children: React.ReactNode}) => {
+  const dispatch = useAppDispatch();
   const darkMode = useAppSelector(selectDarkMode);
 
+  //On the first load, if the system settings are set to dark mode, then set the dark mode to true.
   useEffect(() => {
-    if (darkMode) {
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      document.documentElement.classList.add('dark');
+      localStorage.setItem('parcaDarkModeSystemSettings', 'true');
+      dispatch(setDarkMode(true));
+    }
+  }, [dispatch]);
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+
+    // Listen for changes to the prefers-color-scheme media query and then update the theme accordingly.
+    mediaQuery.onchange = media => {
+      // @ts-expect-error
+      if (media.currentTarget.matches === true) {
+        document.documentElement.classList.add('dark');
+      } else if (
+        // @ts-expect-error
+        media.currentTarget.matches === false
+      ) {
+        document.documentElement.classList.remove('dark');
+      } else {
+        document.documentElement.classList.remove('dark');
+      }
+    };
+
+    if (localStorage['parcaDarkModeSystemSettings']) dispatch(setDarkMode(mediaQuery.matches));
```

Comment:
Instead of manually manipulating the state in `localStorage`, should we move this to the `ui` redux state so that it gets persisted automatically?

Reply:
hmm so I am manipulating the localStorage because we need to know on the next visit to the Parca app, if a user had previously set that they would like to use the system settings to determine the theme mode.

If I were to use the redux state, doesn't that mean, it wouldn't get remembered on the next visit?

Reply:
Oh I see what you mean, Redux's persistence already utilizes localStorage, and I should just use that instead of the localStorage API right?

Reply:
No, I meant not to interact with localStorage at all. 
From this code: https://github.com/parca-dev/parca/blob/4c75077d4951d88f464803d28562e78a40a44512/ui/packages/shared/store/src/store.ts#L39

If I'm not mistaken, we've configured Redux to persist the `ui` slice. So internally, Redux should handle the localStorage saving and restoring logic. 
And if we move `parcaDarkModeSystemSettings` field to the `ui` slice then, we don't have to interact with the localStorage no? Just save it to redux state and it takes care of the persistence.

Reply:
yeah that makes total sense. Thanks for catching that. I'mm refactor that bit.